src/dune_engine/dune_project.ml

```diff
@@ -551,7 +578,118 @@ module Toggle = struct
     field_o name decode
 end
 
-let anonymous ~dir = infer ~dir Package.Name.Map.empty
+let anonymous ~dir ?info ?(packages = Package.Name.Map.empty) () =
+  infer ~dir ?info packages
+
+let encode : t -> Dune_lang.t list =
+ fun { name
+     ; version
+     ; dune_version
+     ; info
+     ; packages
+     ; implicit_transitive_deps
+     ; wrapped_executables
+     ; executables_implicit_empty_intf
+     ; accept_alternative_dune_file_name
+     ; generate_opam_files
+     ; use_standard_c_and_cxx_flags
+     ; dialects
+     ; explicit_js_mode
+     ; format_config
+     ; strict_package_deps
+     ; cram
+     ; subst_config
+       (* The next three fields all get parsed out from the `using` stanza, but
+          we don't need them for project initialization. They should be
+          reconstructed if you ever want a full encoding of the project record
+          tho. *)
+     ; extension_args = _
+     ; parsing_context = _
+     ; stanza_parser =
+         _
+         (* The next three fields hold metadata that is about the dune-project
+            file, but not represented in its content *)
```

Comment:
This is how ocamlformat insisted on formatting this comment. Happy to reformat if given explicit instructions on how you'd prefer it look.

Reply:
The formatting looks good to me. But in general, this is the deal with ocamlformat: the overall benefit is good enough that we accept the occasional formatting we don't like.

Reply:
I agree it's worth it, and I don't mind the slight blemish :)